File: classes/parte-3-valores-tipos-y-variables/041-literales-valores-variables-y-constantes/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	linea, _ := reader.ReadString('\n')
	campos := strings.Fields(linea)

	// Go: tipado estático explícito; conversión float64(cantidad) obligatoria.
	precioUnitario, _ := strconv.ParseFloat(campos[0], 64)
	cantidad, _ := strconv.Atoi(campos[1])
	descuento, _ := strconv.ParseFloat(campos[2], 64)

	subtotal := precioUnitario * float64(cantidad)
	total := subtotal * (1 - descuento)

	fmt.Printf("Total: %.2f\n", total)
}

File: classes/parte-3-valores-tipos-y-variables/041-literales-valores-variables-y-constantes/implementaciones/php/main.php
<?php
// PHP: dinámico y débilmente tipado; las variables llevan el prefijo $.
$linea = trim(fgets(STDIN));
[$precio, $cantidad, $descuento] = preg_split('/\s+/', $linea);

$precioUnitario = (float) $precio;
$cantidadInt = (int) $cantidad;
$descuentoFloat = (float) $descuento;

$subtotal = $precioUnitario * $cantidadInt;
$total = $subtotal * (1 - $descuentoFloat);

printf("Total: %.2f\n", $total);
